python/cmsis_svd/parser.py

```diff
@@ -291,11 +168,122 @@ def preprocess_xml(self):
         self._propagate_register_properties_group()
 
 
-class SVDParser(object):
+def _get_text(node: Element, tag: str, default: Any = None) -> Union[str, Any]:
+    """Get the text for the provided tag from the provided node"""
+    try:
+        return node.find(tag).text
+    except AttributeError:
+        return default
+
+
+def _get_int(node: Element, tag: str, default: Any = None) -> Union[int, Any]:
+    text_value = _get_text(node, tag, default)
+    try:
+        if text_value != default:
+            text_value = text_value.strip().lower()
+
+            if text_value.startswith('0x'):
+                ret_value = int(text_value[2:], 16)  # hexadecimal
+            elif text_value.startswith('#'):
+                # TODO(posborne): Deal with strange #1xx case better
+                #
+                # Freescale will sometimes provide values that look like this:
+                #   #1xx
+                # In this case, there are a number of values which all mean the
+                # same thing as the field is a "don't care".  For now, we just
+                # replace those bits with zeros.
+                text_value = text_value.replace('x', '0')[1:]
+                is_bin = all(x in '01' for x in text_value)
+                ret_value = int(text_value, 2) if is_bin else int(text_value)  # binary
+            elif text_value.startswith('true'):
+                ret_value = 1
+            elif text_value.startswith('false'):
+                ret_value = 0
+            else:
+                ret_value = int(text_value)  # decimal
+
+            return ret_value
+    except ValueError:
+        return default
+    return default
+
+
+def _parse_bool(text_value: str) -> Union[bool, None]:
+    text_bool = text_value.lower().strip()
+
+    if text_bool == '0' or text_bool == 'false':
+        return False
+    elif text_bool == '1' or text_bool == 'true':
+        return True
+
+    return None
+
+
+def _get_bool(node: Element, tag: str, default: Any = None) -> Union[bool, Any]:
+    if text_value := _get_text(node, tag, default):
+        return _parse_bool(text_value)
+    return None
+
+
+class SVDParserValidationError(Exception):
+    pass
+
+
+class SVDParser:
     """The SVDParser is responsible for mapping the SVD XML to Python Objects"""
 
+    SVD_SCHEMA_VERSION = [
+        '1.0', '1.1', '1.2', '1.3.1', '1.3.2', '1.3.3', '1.3.5', '1.3.6',
+        '1.3.7', '1.3.9', '1.3.10',
+    ]
```

Comment:
As this is somewhat hidden and may change as new versions of the SVD standard are released (or other historic versions added ;-)), this might be better off initialized dynamically based on the `.xsd` files available in the package.

Fine as a fallback though.

Reply:
Very good advice, thank you :), fixed.